File: LeetCode Solutions/February 2021 LeetCoding Challenge/Shortest Unsorted Continuous Subarray/Shortest Unsorted Continuous Subarray.cpp
class Solution {
public:
    int findUnsortedSubarray(vector<int>& nums) {
        int lenNums = nums.size()-1, left = -1, right = -1, maxNums = nums[0], minNums = nums[nums.size() - 1];
        for (int i=0; i<nums.size(); i++){
            int a = nums[i], b = nums[lenNums - i];
            if (a < maxNums){
                right = i;
            } else{
                maxNums = a;
            }
            if (b > minNums){
                left = i;
            } else{
                minNums = b;
            }
        }
        return max(0, left + right - lenNums + 1);
    }
};

File: LeetCode Solutions/LeetCoding Challenge/February 2021 LeetCoding Challenge/Shortest Unsorted Continuous Subarray/Shortest Unsorted Continuous Subarray.py
class Solution:
    def findUnsortedSubarray(self, nums: List[int]) -> int:
        lenNums, left, right = len(nums)-1, -1, -1
        maxNums, minNums = nums[0], nums[lenNums]
        for i in range(1, len(nums)):
            a, b = nums[i], nums[lenNums-i]
            if (a < maxNums):
                right = i
            else:
                maxNums = a
            if (b > minNums):
                left = i
            else:
                minNums = b
        return max(0, left + right - lenNums + 1)